graph/R-GAT/tools/accuracy_igbh.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import numpy as np
+import torch
+import json
+import os
+
+
+def get_args():
+    """Parse commandline."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--mlperf-accuracy-file", required=True, help="path to mlperf_log_accuracy.json"
+    )
+    parser.add_argument(
+        "--dataset-path",
+        default="igbh",
+        help="Path to IHGB dataset",
+    )
+    parser.add_argument(
+        "--dataset-size",
+        default="full",
+        choices=["tiny", "small", "medium", "large", "full"]
+    )
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        help="verbose messages")
+    parser.add_argument(
+        "--output-file", default="coco-results.json", help="path to output file"
+    )
+    parser.add_argument(
+        "--dtype",
+        default="int32",
+        choices=["float32", "int32", "int64"],
+        help="data type of the label",
+    )
+    args = parser.parse_args()
+    return args
+
+def load_labels(base_path, dataset_size, use_label_2K = True):
+    # load labels
+    paper_nodes_num = {
+        "tiny": 100000,
+        "small": 1000000,
+        "medium": 10000000,
+        "large": 100000000,
+        "full": 269346174,
+    }
+    label_file = (
+        "node_label_19.npy" if not use_label_2K else "node_label_2K.npy"
+    )
+    paper_lbl_path = os.path.join(base_path, "paper", label_file)
+
+    if dataset_size in ["large", "full"]:
+        paper_node_labels = torch.from_numpy(
+            np.memmap(
+                paper_lbl_path, dtype="float32", mode="r", shape=(paper_nodes_num[dataset_size])
+            )
+        ).to(torch.long)
+    else:
+        paper_node_labels = torch.from_numpy(
+            np.load(paper_lbl_path)).to(
+            torch.long)
+    labels = paper_node_labels
+    val_idx = torch.load(
+                os.path.join(
+                    base_path,
+                "val_idx.pt"))
+    return labels, val_idx
+
+
+def get_labels(labels, val_idx, id_list):
+    return labels[val_idx[id_list]]
+
+
+if __name__ == "__main__":
+    args = get_args()
+    dtype_map = {"float32": np.float32, "int32": np.int32, "int64": np.int64}
+
+    with open(args.mlperf_accuracy_file, "r") as f:
+        mlperf_results = json.load(f)
+
+    labels, val_idx = load_labels(args.dataset_path, args.dataset_size)
+    results = {}
+
+    seen = set()
+    good = 0
+    total = 0
+    for result in mlperf_results:
+        idx = result["qsl_idx"]
+        if idx in seen:
+            continue
+        seen.add(idx)
+
+        # get ground truth
+        label = get_labels(labels, val_idx, idx)
+        # get prediction
+        data = int(np.frombuffer(bytes.fromhex(result["data"]), dtype_map[args.dtype])[0])
```

Comment:
I encountered the following error on this line

```
Traceback (most recent call last):
  File "/home/cmuser/CM/repos/local/cache/ba3921c1b0644c8f/inference/graph/R-GAT/tools/accuracy_igbh.py", line 101, in <module>
    data = int(np.frombuffer(bytes.fromhex(result["data"]), dtype_map[args.dtype])[0])
ValueError: buffer size must be a multiple of element size
```

I tested with the `mlperf_log_accuracy.json` generated from a run with the debug dataset, the element in json file looked like this:
```
{'seq_id': 0, 'qsl_idx': 230, 'data': '09'}
```

It seems that the data key contains a string of length 2, which results in a buffer array of 1 byte. I manually padded the remaining 3 bytes using `result["data"].ljust(8, "0")` as int32 is selected by default, and it worked.

Reply:
I am saving the type in `np.uint8`. I updated the script to use this typing by default as well and that should fix this issue